agent/src/IngestDetailsRepository.php

```diff
@@ -61,62 +70,127 @@ public function get(): PromiseInterface
     private function refresh(): PromiseInterface
     {
         $start = microtime(true);
+        $duration = null;
 
-        return $this->browser->post('', headers: [
-            'user-agent' => 'NightwatchAgent/'.$this->packageVersion->get(),
-        ])->then(function (ResponseInterface $response) use ($start): IngestDetails {
-            $duration = microtime(true) - $start;
+        return $this->browser->post('/api/agent-auth', headers: [], body: '')
+            ->then(function (ResponseInterface $response) use ($start, &$duration): IngestDetails {
+                $duration = microtime(true) - $start;
 
-            $data = json_decode($response->getBody()->getContents(), associative: true, flags: JSON_THROW_ON_ERROR);
+                $ingestDetails = $this->parseResponse($response);
 
-            if (
-                ! is_array($data) ||
-                ! is_string($data['token'] ?? null) ||
-                ! is_int($data['expires_in'] ?? null) ||
-                ! is_string($data['ingest_url'] ?? null)
-            ) {
-                throw new RuntimeException("Invalid authentication response [{$response->getBody()->getContents()}].");
-            }
+                $this->scheduleRefreshIn($ingestDetails->refreshIn);
 
-            $ingestDetails = new IngestDetails($data['token'], $data['expires_in'], $data['ingest_url']);
+                call_user_func($this->onAuthenticationSuccess, $ingestDetails, $duration);
 
-            $this->scheduleRefreshIn(seconds: $ingestDetails->expiresIn - $this->preemptivelyRefreshInSeconds);
+                $this->hasAuthenticated = true;
+                $this->consecutiveFailures = 0;
 
-            call_user_func($this->onAuthenticationSuccess, $ingestDetails, $duration);
+                return $ingestDetails;
+            })->catch(function (Throwable $e) use ($start, &$duration): null {
+                $this->consecutiveFailures++;
 
-            return $ingestDetails;
-        }, function (Throwable $e) use ($start): null {
-            $duration = microtime(true) - $start;
+                // TODO if the current token has expired we should `null` it.
```

Comment:
Are we going to handle it in a separate PR?

Reply:
I plan to. Just wanted to keep the scope small but didn't want to forget about it!